src/commonMain/kotlin/com/github/quillraven/fleks/entity.kt

```diff
@@ -276,7 +287,7 @@ class DefaultEntityProvider(
                 nextId = recycled.id + 1
             }
 
-            recycled
+            recycled.copy(version = ++currentEntityVersions[recycled.id])
```

Comment:
Similar to my comment for the `Entity data class`, we are creating here a new object instance every  time an entity gets recycled. 
I understand why we need to do that because we cannot modify the original instance because otherwise the entity refs in components will not get invalid, but again it creates additional object instances compared to current Fleks.